pkg/topo/client/client.go

```diff
@@ -0,0 +1,177 @@
+// SPDX-FileCopyrightText: 2020-present Open Networking Foundation <info@opennetworking.org>
+//
+// SPDX-License-Identifier: LicenseRef-ONF-Member-1.0
+
+package client
+
+import (
+	"context"
+
+	"github.com/onosproject/onos-ric-sdk-go/pkg/topo/options"
+
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+
+	"io"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+	"google.golang.org/grpc/status"
+
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+
+	"google.golang.org/grpc"
+)
+
+var log = logging.GetLogger("topo", "client")
+
+// Topo is a topo client interface
+type Topo interface {
+	// Create creates an R-NIB object
+	Create(ctx context.Context, object *topoapi.Object) error
+
+	// Update updates an existing R-NIB object
+	Update(ctx context.Context, object *topoapi.Object) error
+
+	// Get gets an R-NIB object
+	Get(ctx context.Context, id topoapi.ID) (*topoapi.Object, error)
+
+	// List lists R-NIB objects
+	List(ctx context.Context, opts options.ListOptions) ([]topoapi.Object, error)
+
+	// Delete deletes an R-NIB object using the given ID
+	Delete(ctx context.Context, id topoapi.ID) error
+
+	// Watch watches topology events
+	Watch(ctx context.Context, ch chan<- topoapi.Event, opts options.WatchOptions) error
```

Comment:
I know the Manager interface has variadic methods, but I think we should pass them through and make the client interface methods variadic as well.